src/number_theory/nat_prop.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2021 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan
+-/
+import data.nat.parity
+import ring_theory.int.basic
+
+/-!
+# Properties of natural numbers
+This file describes some properties of natural numbers.
+
+## Tags
+Naturals
+-/
+
+namespace nat
+lemma coprime.sub {n m : ℕ} (h : n.coprime m) (hn : m ≤ n) : (n - m).coprime n :=
+begin
+  by_contradiction h',
+  obtain ⟨p, h1, h2, h3⟩ := nat.prime.not_coprime_iff_dvd.1 h',
+  have h4 := nat.dvd_sub (nat.sub_le _ _) h3 h2,
+  rw nat.sub_sub_self hn at h4,
+  apply nat.prime.not_coprime_iff_dvd.2 ⟨p, h1, h3, h4⟩ h,
+end
+
+lemma lt_mul_pow_right {m a b : ℕ} (h1 : 0 < b) (h2 : 1 < a) (h3 : 1 < m) : a < b * a^m :=
+lt_of_le_of_lt ((le_mul_iff_one_le_left (lt_trans zero_lt_one h2)).2 h1)
+  (mul_lt_mul' le_rfl (lt_pow h2 h3) (nat.zero_le _) h1)
+
+lemma le_mul_pow_right {m a b : ℕ} (h1 : 0 < b) (h2 : 1 < a) (h3 : 1 ≤ m) : a ≤ b * a^m :=
+le_trans ((le_mul_iff_one_le_left (lt_trans zero_lt_one h2)).2 h1)
+  (mul_le_mul' le_rfl (le_pow (le_of_lt h2) h3))
+
+lemma cast_eq_coe_b (x : ℕ) : @nat.cast ℤ _ x = coe_b x :=
+begin
+  induction x with d hd,
+  { change 0 = @has_coe.coe ℕ ℤ _ 0,
+    change _ = int.of_nat 0,
+    simp only [int.coe_nat_zero, int.of_nat_eq_coe], },
+  { show d.cast + 1 = @has_coe.coe ℕ ℤ _ d.succ,
+    change _ = int.of_nat d.succ,
+    simp only [int.of_nat_eq_coe, int.coe_nat_succ, add_left_inj],
+    change _ = int.of_nat d at hd, simp [hd], },
+end
```

Comment:
This lemma looks very strange to me; why do you need it; and isn't the proof just `rfl`?

Reply:
Turns out I don't need it, so I will remove it. The proof isn't `rfl`, by the way.